src/qibolab/transpilers/abstract.py

```diff
@@ -1,12 +1,57 @@
 from abc import ABC, abstractmethod
 from typing import List, Optional, Tuple
 
+import networkx as nx
 import numpy as np
 from qibo.backends import NumpyBackend
-from qibo.config import log
+from qibo.config import log, raise_error
 from qibo.models import Circuit
 
 
+# TODO update circuit representation
+def create_circuit_repr(circuit):
+    """Translate qibo circuit into a list of two qubit gates to be used by the transpiler.
+
+    Args:
+        circuit (:class:`qibo.models.Circuit`): circuit to be transpiled.
+
+    Returns:
+        translated_circuit (list): list containing qubits targeted by two qubit gates
+    """
+    translated_circuit = []
+    index = 0
+    for gate in circuit.queue:
+        if len(gate.qubits) == 2:
+            gate_qubits = list(gate.qubits)
+            gate_qubits.sort()
+            gate_qubits.append(index)
+            translated_circuit.append(gate_qubits)
+            index += 1
+        if len(gate.qubits) >= 3:
+            raise_error(ValueError, "Gates targeting more than 2 qubits are not supported")
+    return translated_circuit
+
+
+class Placer(ABC):
+    """A placer implments the initial logical-physical qubit mapping"""
+
+    @abstractmethod
+    def __init__(self, connectivity: nx.Graph, *args):
+        """"""
```

Comment:
For `__init__` you can also save the docstring: there would not be much difference between the docstring in here, and class one, so better to keep only one :)